public_html/wp-content/mu-plugins/blocks/assets/src/sponsors/block-controls.js

```diff
@@ -0,0 +1,345 @@
+				const sponsorTermOrder = fetchedSettings.wcb_sponsor_level_order;
+				this.setState( { sponsorTermOrder } );
+			}
+		);
+
+		Promise.all( [ parsedPosts, parsedTerms, parsedSettings ] ).then( () => {
+			this.setState( { loading: false } );
+			// Enqueue set selected posts in next event loop, so that state is up to date when `setSelectedPosts` method actually runs.
+			setTimeout( () => this.setSelectedPosts() );
+		} );
+	}
+
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
+	/**
+	 * Sets `mode`, `term_ids` and `post_ids` attribute when `Apply` button is
+	 * clicked. Pass `onChange` prop to override.
+	 *
+	 * @param selectedOptions Array of values, type of selected options
+	 */
+	onChange( selectedOptions = {} ) {
+		const { setAttributes } = this.props;
+		const newValue = selectedOptions.item_ids;
+		const chosen = selectedOptions.mode;
+
+		if ( newValue && chosen ) {
+
+			switch ( chosen ) {
+				case 'post' :
+					setAttributes( {
+						mode     : 'specific_posts',
+						post_ids : newValue,
+					} );
+					break;
+
+				case 'term' :
+					setAttributes( {
+						mode     : 'specific_terms',
+						term_ids : newValue,
+					} );
+					break;
+			}
+		} else {
+			setAttributes( {
+				mode     : '',
+				post_ids : [],
+				term_ids : [],
+			} );
+		}
+
+		setTimeout( () => this.setSelectedPosts() );
+	}
+
+	/**
+	 * Generate options array to be passed to select2.
+	 */
+	buildSelectOptions() {
+		const { posts, terms } = this.state;
+		const options = [];
+
+		options.push( {
+			label   : __( 'Sponsor Levels', 'wordcamporg' ),
+			options : terms,
+		} );
+
+		options.push( {
+			label   : __( 'Sponsors', 'wordcamporg' ),
+			options : posts,
+		} );
+
+		return options;
+	}
+
+	/**
+	 * Renders Sponsor Block Control view
+	 */
+	render() {
+		const { sponsorPosts, attributes, setAttributes } = this.props;
+		const { mode, post_ids, term_ids } = attributes;
+		const { fetchedPosts, posts, terms, selectedPosts, sponsorTermOrder } = this.state;
+		const hasPosts = Array.isArray( fetchedPosts ) && fetchedPosts.length;
+
+		// Check if posts are still loading.
+		if ( mode && ! hasPosts ) {
+			return (
+				<PlaceholderNoContent
+					label = { LABEL }
+					loading = { () => {
+						return ! Array.isArray( sponsorPosts );
+					} }
+				/>
+			)
+		}
+
+		let selectedOptions = [];
+
+		switch ( mode ) {
+			case 'all' :
+				break;
+			case 'specific_posts' :
+				selectedOptions = posts.filter( ( post ) => {
+					return includes( post_ids, post.value );
+				} );
+				break;
+			case 'specific_terms' :
+				selectedOptions = terms.filter( ( term ) => {
+					return includes( term_ids, term.value );
+				} );
+				break;
+			default:
+				break;
+		}
+
+		return (
+			<div>
+
+				<SponsorBlockContent
+					selectedPosts={ selectedPosts }
+					sponsorTermOrder = { sponsorTermOrder }
+					{...this.props}
+				/>
+
+				{'all' !== mode &&
+					<Placeholder
+						icon='heart'
```

Comment:
`[MUST]` - @vedanshujain , what's the reason that this is done differently than Speakers? Here it's hardcoded instead of pulling in the `icon` from `this.props`.

It seems like it'd be better to have it DRY like Speakers, but the larger issue is the inconsistency. I think in these cases we need to discuss the two approaches, come to a consensus on the best one, and update all the blocks to use it, so that everything is consistent.

In cases where something is clearly better and it's unlikely that there'd be any disagreement, it's also totally fine to just go ahead and adopt it (since we can always revert if there is a disagreement and we change our minds), but it's still important to update the other components to be consistent, and then give everyone a heads up.

In this case, I'm working on updating the icons per https://github.com/WordPress/wordcamp.org/pull/33#discussion_r267058114, so I'm gonna go ahead and change this to be consistent w/ the others. If there's a reason to change them all to this approach, though, we can definitely start a discussion about that.

Reply:
Opened issue for this https://github.com/WordPress/wordcamp.org/issues/43